src/backend/distributed/executor/executor.c

```diff
@@ -706,7 +708,7 @@ DistributedExecutionRequiresRollback(DistributedExecution *execution)
 		return false;
 	}
 
-	if (execution->operation == CMD_SELECT)
+	if (operation == CMD_SELECT)
 	{
 		return SelectOpensTransactionBlock && IsTransactionBlock();
```

Comment:
Should we ignore this GUC when there is FOR UPDATE?

Reply:
I guess in general without a transaction block, `SELECT FOR UPDATE` doesn't make much sense. Am I missing anything here?

I considered/tests changing here as well, but then (a) we'd be slightly converging from how Postgres behaves (b) could slightly confuse the users. 

So, my slight preference is to keep as-is. But if you prefer to change, I can do that.

Reply:
The GUC is specifically meant for situations where you do `BEGIN; SELECT`. We give the users to not open a transaction block to reduce idle in transaction time, but we should never do that in case of FOR UPDATE.

Reply:
Oh, do you mean ignore the GUC in the sense that if we're in a TX block, have `SELECT FOR UPDATE` and the GUC is OFF, we still open the tx block on the workers?

Reply:
right

Reply:
This makes a lot of sense.